src/test/java/org/stellar/sdk/BumpFootprintExpirationOperationTest.java

```diff
@@ -0,0 +1,111 @@
+package org.stellar.sdk;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNull;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BumpFootprintExpirationOperationTest {
+  @Test
+  public void testBuilderWithSource() {
+    String source = "GASOCNHNNLYFNMDJYQ3XFMI7BYHIOCFW3GJEOWRPEGK2TDPGTG2E5EDW";
+    BumpFootprintExpirationOperation op =
+        BumpFootprintExpirationOperation.builder()
+            .ledgersToExpire(123)
+            .sourceAccount(source)
+            .build();
+    assertEquals(Integer.valueOf(123), op.getLedgersToExpire());
+    assertEquals(source, op.getSourceAccount());
+    String expectXdr = "AAAAAQAAAAAk4TTtavBWsGnEN3KxHw4Ohwi22ZJHWi8hlamN5pm0TgAAABkAAAAAAAAAew==";
```

Comment:
are there other tests in this repo that hardcode expected xdr, that may not be a good pattern to follow, can make the test fragile as the xdr may change, expand over time. Instead, can do a full round trip with marshaling such as `secondOp = Operation.fromXdr(op.toXdrBase64())` and then `assertEquals(secondOp, op)`

Reply:
I tend to keep it. Our XDR module code lacks sufficient testing, and these XDR values are generated by other SDKs (Python, JS), which can serve as cross-validation. Additionally, changes in XDR definitions do not occur frequently, so even if we update it with some effort, it is acceptable.

In the previous code, there were also some validations done in this way. https://github.com/stellar/java-stellar-sdk/blob/master/src/test/java/org/stellar/sdk/OperationTest.java#L167

Reply:
yes, I have seen the usage of hardcoded xdr for serialization tests, not suggesting to change either, but that is more for scope in the xdrgen library to verify. This unit test's responsibility is to validate the code from this sdk builds a valid java object graph for BumpFootprint, and it has asserted that, it doesn't seem like the extra assertions on serialized xdr are related to code in this repo. But, I am relatively newer to this repo, and would be great if @tamirms wants to jump in with any thoughts on this also.